### PR TITLE
Broadcast events

### LINKED ISSUE #221: Broadcast events
Events are displayed when doing `kubectl describe`, and are a useful way of seeing what happened historically to the object and its sub-resources, without having to scan the operator logs.

They can also be displayed with `kubectl get events`.

Events we might emit:

* `SuccessfulCreate` for the `StatefulSet`, `ConfigMap`, `Secret`, `PV`, `PVC` (the STS controller only does it for `Pod`s though)
* Failure for any of the above
* Habitat synced
* Validation failures (this should be removed once #185 is implemented)

### PR DESCRIPTION
**Pull request by asymmetric**
Emits an event when:

* A `Habitat` fails to validate (will have to be removed when #185 is implemented, because an invalid `Habitat` shouldn't even be creatable in the first place)
* One of the sub-resources fails to be created
* A `Habitat` is synced successfully (not sure we **really** need this)

To test: create a `Habitat` and type `kubectl describe hab`

Closes #221.

### CODE REVIEW

**Comment by asymmetric** [comment]
OK, removed the sync event, PTALA.

File: pkg/controller/v1beta2/controller.go [hunk 1]
```diff
@@ -556,16 +594,19 @@ func (hc *HabitatController) conform(key string) error {
 				return err
 			}
 		} else {
+			hc.recorder.Event(h, apiv1.EventTypeWarning, stsFailed, messageStsCreated)
```
Comment (iaguis) [inline pkg/controller/v1beta2/controller.go]: Shouldn't this be `messageStsFailed`?

File: pkg/controller/v1beta2/controller.go [hunk 2]
```diff
@@ -450,9 +484,11 @@ func (hc *HabitatController) handleConfigMap(h *habv1beta1.Habitat) error {
 			return err
 		}
 
-		level.Info(hc.logger).Log("msg", "updated peer IP in ConfigMap", "name", cm.Name, "ip", leaderIP)
+		level.Info(hc.logger).Log("msg", messagePeerIPUpdated, "name", cm.Name, "ip", leaderIP)
+		hc.recorder.Event(h, apiv1.EventTypeNormal, cmUpdated, messagePeerIPUpdated)
 	} else {
-		level.Info(hc.logger).Log("msg", "created peer IP ConfigMap", "name", cm.Name, "ip", leaderIP)
+		level.Info(hc.logger).Log("msg", messageCMCreated, "name", cm.Name, "ip", leaderIP)
+		hc.recorder.Event(h, apiv1.EventTypeNormal, cmUpdated, messagePeerIPAdded)
```
Comment (iaguis) [inline pkg/controller/v1beta2/controller.go]: Shouldn't this be a cm created event?

File: pkg/controller/v1beta2/controller.go [hunk 3]
```diff
@@ -405,12 +437,14 @@ func (hc *HabitatController) handleConfigMap(h *habv1beta1.Habitat) error {
...
-			level.Debug(hc.logger).Log("msg", "removed peer IP from ConfigMap", "name", newCM.Name)
+			level.Debug(hc.logger).Log("msg", messagePeerIPRemoved, "name", newCM.Name)
+			hc.recorder.Event(h, apiv1.EventTypeNormal, cmUpdated, messagePeerIPRemoved)
 
 			return nil
 		}
 
-		level.Info(hc.logger).Log("msg", "created peer IP ConfigMap", "name", cm.Name)
+		level.Info(hc.logger).Log("msg", messageCMCreated, "name", cm.Name)
+		hc.recorder.Event(h, apiv1.EventTypeNormal, cmUpdated, messagePeerIPAdded)
```
Comment (iaguis) [inline pkg/controller/v1beta2/controller.go]: Shouldn't this be a cm created event?

**Comment by iaguis** [comment]
I guess you need something like

```
...
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - get
  - update
  - patch
  - create
```

In the operator RBAC rules

**Pull request merged**